Client/Application/AppLaunchUtil.swift

```diff
@@ -83,27 +83,26 @@ class AppLaunchUtil {
 
     func setUpPostLaunchDependencies() {
 
-        let persistedCurrentVersion = InstallType.persistedCurrentVersion()
-        let introScreen = profile.prefs.intForKey(PrefsKeys.IntroSeen)
-        // upgrade install - Intro screen shown & persisted current version does not match
-        if introScreen != nil && persistedCurrentVersion != AppInfo.appVersion {
-            InstallType.set(type: .upgrade)
-            InstallType.updateCurrentVersion(version: AppInfo.appVersion)
-        }
-
+        // Ecosia: Removing unneded `introScreen` constant
+        // let introScreen = profile.prefs.intForKey(PrefsKeys.IntroSeen)
+        
+        // Ecosia: InstallType logic review
+        
         // We need to check if the app is a clean install to use for
-        // preventing the What's New URL from appearing.
-        if introScreen == nil {
-            // fresh install - Intro screen not yet shown
+        // preventing the What's New Page from appearing.
+        if InstallType.get() == .unknown {
             InstallType.set(type: .fresh)
-            InstallType.updateCurrentVersion(version: AppInfo.appVersion)
-            // Profile setup
-            profile.prefs.setString(AppInfo.appVersion, forKey: LatestAppVersionProfileKey)
-
+            InstallType.updateCurrentVersion(version: AppInfo.ecosiaAppVersion)
+            profile.prefs.setString(AppInfo.ecosiaAppVersion, forKey: LatestAppVersionProfileKey)
         } else if profile.prefs.boolForKey(PrefsKeys.KeySecondRun) == nil {
             profile.prefs.setBool(true, forKey: PrefsKeys.KeySecondRun)
         }
 
+        if InstallType.persistedCurrentVersion() != AppInfo.ecosiaAppVersion {
```

Comment:
I guess this logic here is also persisting the version, somehow similar to what is already done on your newly created component. (but maybe more generic and enabled to be used across other features?)

Just wondering, couldn't this then be used inside that logic as well avoiding repeating it?

Reply:
I can be, yeah. Although the purpose is different. Here we are checking solely the single current version, while with the implementation of `User.shared.whatsNewItemsVersionsShown` we are making sure that (in case of the WhatsNewPage needs showing) only the needed items are delivered against a specific version.

Reply:
You're right, we do have a different logic inside `User` for storing multiple version, I was mostly comparing this logic here with what we have inside `WhatsNewLocalDataProvider.evaluateVersionNeedsUpdate()` since it seems similar with comparing with the current and persisting if updated.